### PR TITLE
tests: fix unit tests and create typescript tests

### PR DESCRIPTION
**Pull request by straker**
The unit tests weren't actually working correctly as we didn't make them `async` tests and use the `done` function. Updated them to work and fixed a failing test. Also added a typescript test so we can verify the input and output to our API.

Lastly, added an axe-linter ignore file so we don't have to see the release PR get axe-linter bugs on the Changelog.

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

File: test/index.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import reactAxe from '../';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+// default use
+reactAxe(React, ReactDOM, 1000);
```
Comment (stephenmathieson) [inline test/index.ts:6]: I don't believe the timeout is _required_. We should also have something like:

```ts
reactAxe(React, ReactDOM)
```

File: test/index.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    {
+      id: 'my-rule',
+      any: ['my-check']
+    }
+  ]
+});
+
+const context = document.createElement('div');
+context.setAttribute('id', 'container');
+document.body.appendChild(context);
```
Comment (stephenmathieson) [inline test/index.ts:26]: These lines aren't doing anything and can be removed.

```suggestion
```

**Comment by straker** [comment]
Yes

File: test/index.ts (hunk 1 above)
Comment (straker) [reply test/index.ts:6]: I thought so too but if you use the option you have to pass it and I didn't feel like updating the code to make that work https://github.com/dequelabs/react-axe/pull/171#issuecomment-647587840

Comment (stephenmathieson) [reply test/index.ts:6]: Interesting, OK. I guess the timeout is required then. We should probably make it an optional parameter with a sensible default in a future release.

**Pull request merged**